File: src/main/java/com/blogly/blogly/domain/post/Content.java
package com.blogly.blogly.domain.post;

public record Content(String value) {

    private static final int MAX_LENGTH = 1_000;

    public Content {
        if (value == null || value.isBlank()) {
            throw new IllegalArgumentException("Content value cannot be null or blank");
        }

        if (value.length() > MAX_LENGTH) {
            throw new IllegalArgumentException("Content value cannot exceed " + MAX_LENGTH + " characters");
        }
    }
}

File: src/main/java/com/blogly/blogly/domain/post/Content.kt
package com.blogly.blogly.domain.post

@JvmInline
value class Content(val value: String) {
    init {
        require(value.isNotBlank()) { "Content cannot be blank" }
        require(value.length <= MAX_LENGTH) { "Content value cannot exceed $MAX_LENGTH characters" }
    }

    companion object {
        private const val MAX_LENGTH = 1000
    }
}
